pkg/asset/installconfig/gcp/validation.go

```diff
@@ -336,3 +337,51 @@ func validateCredentialMode(client API, ic *types.InstallConfig) field.ErrorList
 
 	return allErrs
 }
+
+func validateMarketplaceImages(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	const errorMessage string = "could not find the boot image: %v"
+	const unspecifiedArch string = "ARCHITECTURE_UNSPECIFIED"
+
+	if ic.GCP.DefaultMachinePlatform != nil && ic.GCP.DefaultMachinePlatform.OSImage.Name != "" {
+		osImage := ic.GCP.DefaultMachinePlatform.OSImage
+		instance, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "gcp", "defaultMachinePlatform", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+		}
+		if instance.Architecture == unspecifiedArch {
+			logrus.Warn(fmt.Sprintf("Image architecture is unspecified for project '%s' and image '%s'", osImage.Name, osImage.Project))
+		}
+	}
+
+	if ic.ControlPlane != nil && ic.ControlPlane.Platform.GCP != nil && ic.ControlPlane.Platform.GCP.OSImage.Name != "" {
+		osImage := ic.ControlPlane.Platform.GCP.OSImage
+		instance, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("controlPlane", "platform", "gcp", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+		}
+		if instance.Architecture == unspecifiedArch {
+			logrus.Warn(fmt.Sprintf("Image architecture is unspecified for project '%s' and image '%s'", osImage.Name, osImage.Project))
+		}
```

Comment:
This is not enough to validate the boot image. We don't want to use an `arm64` image for `amd64` controlPlane nodes. So we need something like:
```suggestion
		if image.Architecture != ic.ControlPlane.Architecture {
		    if image.Architecture == unspecifiedArch {
			    logrus.Warn(fmt.Sprintf("Boot image architecture is unspecified and might not be compatible with %s controlPlane nodes", ic.ControlPlane.Architecture))
			} else {
			    errMsg := fmt.Sprintf("image architecture %s does not match node architecture %s", image.Architecture, ic.ControlPlane.Architecture)
			    allErrs = append(allErrs, field.Invalid(field.NewPath("controlPlane", "platform", "gcp", "osImage"), osImage, errMsg)))
			}
		}
```
Unfortunately, it's a bit more complicated than that since, if `osImage` is not specified for `controlPlane`. we use the `osImage` from the `defaultMachinePlatform`.

Reply:
Also bear in mind that `image.Architecture` is not directly comparable to `ic.ControlPlane.Architecture` since GCP uses `ARM64` and `X86_64`, whereas the installer uses `arm64` and `amd64`.